lib/Service/CardService.php

```diff
@@ -390,6 +391,29 @@ public function update($id, $title, $stackId, $type, $owner, $description = '',
 		return $card;
 	}
 
+	public function cloneCard(int $id, ?int $targetStackId = null):Card {
+		$this->permissionService->checkPermission($this->cardMapper, $id, Acl::PERMISSION_READ);
+		$originCard = $this->cardMapper->find($id);
+		if ($targetStackId == null) {
+			$targetStackId = $originCard->getStackId();
+		}
+		$this->permissionService->checkPermission($this->stackMapper, $targetStackId, Acl::PERMISSION_EDIT);
+		$newCard = $this->create($originCard->getTitle(), $targetStackId, $originCard->getType(), $originCard->getOrder(), $originCard->getOwner());
+		$boardId = $this->stackMapper->findBoardId($targetStackId);
+		foreach ($this->labelMapper->findAssignedLabelsForCard($id) as $label) {
+			if ($boardId != $this->stackMapper->findBoardId($originCard->getStackId())) {
+				$label = $this->labelService->cloneLabelIfNotExists($label->getId(), $boardId);
```

Comment:
A user is only able to add/edit labels on a board with manage permission iirc so we should also take that into account when cloning the labels

Reply:
Ah it is checked in https://github.com/nextcloud/deck/pull/6452/files#diff-28de0536daa5474c78d675a3bd3c4ab7a9d26f4de82031e5b16bf89dbdc30f3cR96, then let's catch the exception here and skip the label cloning I'd say